shared/common-adapters/button.desktop.js

```diff
@@ -2,40 +2,124 @@
         label={this.props.label || this.props.more && '•••'}
-        primary={this.props.primary} />
+        primary={this.props.type === 'Primary'}
+        secondary={this.props.type === 'Secondary'}
+      />
     )
   }
-}
 
-Button.propTypes = {
-  onClick: React.PropTypes.func.isRequired,
-  label: React.PropTypes.string,
-  style: React.PropTypes.object,
-  primary: React.PropTypes.bool,
-  small: React.PropTypes.bool,
-  more: React.PropTypes.bool
+  render () {
+    if (!flags.tracker2) {
+      return this.render1()
+    }
+
+    let backgroundStyle = {}
+    let labelStyle = {}
+    let smallStyle = {}
+
+    // First apply styles for the main button types.
+    switch (this.props.type) {
+      case 'Primary':
+        backgroundStyle = {
+          ...styles2.buttonPrimary,
+          opacity: this.props.disabled ? styles2.buttonPrimary.disabledOpacity : 1
+        }
+        break
+      case 'Follow':
+        backgroundStyle = {
+          ...styles2.buttonFollow,
+          opacity: this.props.disabled ? styles2.buttonFollow.disabledOpacity : 1
+        }
+        break
+      case 'Following':
+        backgroundStyle = {
+          ...styles2.buttonFollowing,
+          opacity: this.props.disabled ? styles2.buttonFollowing.disabledOpacity : 1
+        }
+        labelStyle = {
+          color: globalColorsDZ2.green
+        }
+        break
+      case 'Unfollow':
+        backgroundStyle = {
+          ...styles2.buttonUnfollow,
+          opacity: this.props.disabled ? styles2.buttonUnfollow.disabledOpacity : 1
+        }
+        break
+      case 'Danger':
+        backgroundStyle = {
+          ...styles2.buttonDanger,
+          opacity: this.props.disabled ? styles2.buttonDanger.disabledOpacity : 1
+        }
+        break
+      case 'Secondary':
+      default:
+        backgroundStyle = {
+          ...styles2.buttonSecondary,
+          opacity: this.props.disabled ? styles2.buttonSecondary.disabledOpacity : 1
+        }
+        labelStyle = {
+          color: globalColorsDZ2.black75
+        }
+    }
+
+    // Then some overrides that apply to all button types.
+    if (this.props.small) {
+      smallStyle = styles2.buttonSmall
+      labelStyle = {
+        ...labelStyle,
+        ...styles2.buttonSmallLabel
+      }
+    }
+
+    if (this.props.fullWidth) {
+      backgroundStyle = {
+        ...backgroundStyle,
+        // Using minWidth here means we can't have a full-width button on the
+        // same line/row as another button, the right thing is very unlikely to
+        // happen.  The alternative is 'flex: 1' here, which would work but is
+        // dangerous, because we'd be modifying our container.
+        //
+        // So let's just say that a fullWidth button can't have siblings.
+        minWidth: '100%'
+      }
+    }
+
+    return (
+      <FlatButton
+        onClick={this.props.onClick}
+        style={{...backgroundStyle, ...this.props.style, ...smallStyle}}
```

Comment:
why smallStyle after this.props.style,  i assume you want this.props.style last so it actually overrides everything